Purchases/Public/Purchases.swift

```diff
@@ -993,18 +995,17 @@ public extension Purchases {
      */
     @available(iOS 12.2, macOS 10.14.4, watchOS 6.2, macCatalyst 13.0, tvOS 12.2, *)
     @objc(purchasePackage:withDiscount:completion:)
-    func purchase(package: Package, discount: SKPaymentDiscount, completion: @escaping PurchaseCompletedBlock) {
+    func purchase(package: Package, discount: StoreProductDiscount, completion: @escaping PurchaseCompletedBlock) {
         // todo: add support for SK2 with discounts, move to new class
         // https://github.com/RevenueCat/purchases-ios/issues/1055
         guard let sk1Product = package.storeProduct.sk1Product else {
             return
         }
-        let payment = storeKitWrapper.payment(withProduct: sk1Product,
-                                              discount: discount)
-        purchase(product: sk1Product,
-                 payment: payment,
-                 presentedOfferingIdentifier: package.offeringIdentifier,
-                 completion: completion)
+
+        purchasesOrchestrator.purchase(sk1Product: sk1Product,
+                                       storeProductDiscount: discount,
+                                       presentedOfferingIdentifier: package.offeringIdentifier,
```

Comment:
I'm wondering if it could be better to pass the package in instead of `package.offeringIdentifier`

Reply:
Oh much better idea! Love it